File: readability/readability.c
#include <stdio.h>
#include <string.h>
#include <cs50.h>
#include <ctype.h>
#include <math.h>

float grade (string text);

int main (void)
{
    string text = get_string("Text: ");

    float index = grade(text);

    if (index < 1)
        printf("Before Grade 1\n");
    else if (index >= 16)
        printf ("Grade 16+\n");
    else
        printf ("Grade %i\n", (int)round(index));

}

float grade (string text)
{
    int letters = 0;
    int words = 1;
    int sentences = 0;
    float L, S;
    float index;

    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
            letters++;
        if (text[i] == ' ')
            words++;
         if (text[i] == '.' || text[i] == '?' || text[i] == '!')
            sentences++;
    }
    L = (float)(100.0 * letters) / (float)words;
    S = (float)(100.0 * sentences) / (float)words;

    index = 0.0588 * L - 0.296 * S - 15.8;
    return index;
}

File: sentimental-readability/readability.py
from cs50 import get_string

def main ():
    text = input("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    L = 100*letters/words
    S = 100*sentences/words
    index = round(0.0588 * L - 0.296 * S - 15.8)
    if index<1:
        print("Before Grade 1")
    if index>16:
        print("Grade 16+")
    elif(index > 1 and index < 16):
        print(f"Grade: {index}" )


def count_letters(text):
    counter = 0
    for char in text:
        if char.isalpha():
            counter += 1

    return counter



def count_words(text):
    counter = 1
    for i in range (0,len(text)):
        if text[i] == " ":
            counter+= 1

    return counter

def count_sentences(text):
    counter = 0
    for i in range(0,len(text)):
        if text[i] in ["?",".","!"]:
            counter += 1
    return counter


main()


